### PR TITLE
[pulsar-broker] Added support to force deleting tenant

### PR DESCRIPTION
**Pull request by murong00**
### Motivation

Some users expect an option to force deleting tenant for simplicity, it can be useful in some situations.

### Modifications

Add a optional field to force the deletion of all stuffs related to tenant and a related unit test.

### CODE REVIEW

**Comment by jiazhai** [comment]
@BewareMyPower Would you please also take a look of this PR?

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest.java [hunk 1]
```diff
@@ -1108,6 +1108,46 @@ public void testDeleteNamespaceBundle(Integer numBundles) throws Exception {
...
+        assertFalse(admin.topics().getList(namespace).isEmpty());
+
+        try {
+            admin.tenants().deleteTenant(tenant, false);
+            fail("should have failed");
+        } catch (PulsarAdminException e) {
+            // Expected: cannot delete non-empty tenant
+        }
+
+        //
```
Comment (BewareMyPower) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest.java:1141]: What is your intention for this comment?

Comment (murong00) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest.java:1141]: Just want to comment "delete tenant forcefully", sorry for the omission.

Comment (murong00) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest.java]: Done.

**Review by BewareMyPower — APPROVED** [review]
LGTM

**Comment by Anonymitaet** [comment]
@murong00 thanks for your coding work, would you like to add docs for this change? I'd like to help review. Thanks

**Comment by murong00** [comment]
> @murong00 thanks for your coding work.
> 
> 1. Does this apply to 2.7.1?
> 2. Would you like to add docs for this change?

@Anonymitaet 
1. Applying this to 2.8.0 seems more appropriate since it is a new feature.
2. Done, please help to take a look.

File: site2/docs/reference-pulsar-admin.md [hunk 2]
```diff
@@ -2451,6 +2451,11 @@ Usage
 $ pulsar-admin tenants delete tenant-name
 ```
 
+Options
+|Flag|Description|Default|
+|----|---|---|
+|`-f`, `--force`|Delete tenant forcefully by force deleting all namespaces under it|false|
```
Comment (Anonymitaet) [inline site2/docs/reference-pulsar-admin.md:2457]:
```suggestion
|`-f`, `--force`|Delete a tenant forcefully by deleting all namespaces under it. |false|
```

**Review by Anonymitaet — COMMENTED** [review]
Thanks. Please update the description for all occurrences.

File: site2/docs/reference-pulsar-admin.md (hunk 2 above)
Comment (murong00) [reply site2/docs/reference-pulsar-admin.md]: Done.

**Review by Anonymitaet — APPROVED** [review]

**Comment by murong00** [comment]
@codelipenghui Please help to take a look about this pr, thanks.

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@murong00 Then change looks good to me, shall we need to control the force delete tenants or namespaces in the broker.conf? such as forceDeleteNamepaceAllowed, forceDeleteTenantAllowed? This will help the Pulsar maintainer to maintain the cluster as expected. If this makes sense, we can create an issue or send a PR directly.

I will merge this PR first.

**Comment by murong00** [comment]
> @murong00 Then change looks good to me, shall we need to control the force delete tenants or namespaces in the broker.conf? such as forceDeleteNamepaceAllowed, forceDeleteTenantAllowed? This will help the Pulsar maintainer to maintain the cluster as expected. If this makes sense, we can create an issue or send a PR directly.

That sounds like a good idea, I will accomplish this when available.

**Comment by codelipenghui** [comment]
@murong00 Ok, I will create an issue first

**Pull request merged**